src/Test/Integration/FeatureContextAbstract.php

```diff
@@ -0,0 +1,235 @@
+     * @var string
+     */
+    protected $calledUrl;
+
+    /**
+     * @return DbInitializerAbstract
+     */
+    abstract protected function getDbInitializer();
+
+    /**
+     * @return array
+     */
+    abstract protected function getLogResourceNames();
+
+    /**
+     * @return array
+     */
+    abstract protected function getStorageNames();
+
+    /**
+     * @BeforeScenario
+     */
+    public function initScenario()
+    {
+        $this->getDbInitializer()->initDatabase();
+        $this->initLogs();
+        $this->cleanUpStorages();
+    }
+
+    /**
+     * @AfterScenario
+     * @throws \YapepBase\Exception\Exception
+     */
+    public function afterScenario()
+    {
+        $loggedErrors = $this->getLoggedErrors();
+
+        if (!empty($loggedErrors)) {
+            throw new Exception('Errors logged during the test: ' . PHP_EOL . implode(PHP_EOL, $loggedErrors));
+        }
+    }
+
+    public function initLogs()
+    {
+        $config = Config::getInstance();
+        foreach ($this->getLogResourceNames() as $resourceName) {
+            $errorLogPath = $config->get('resource.log.' . $resourceName . '.path');
+
+            $fileHandler = new FileHandlerPhp();
+
+            if ($fileHandler->checkIsPathExists($errorLogPath)) {
+                $fileHandler->remove($errorLogPath);
+            }
+        }
+    }
+
+    /**
+     * Cleans the given storage.
+     *
+     * @return void
+     */
+    protected function cleanUpStorages()
+    {
+        foreach ($this->getStorageNames() as $storageName) {
+            StorageFactory::get($storageName)->clear();
+        }
+    }
+
+
+    protected function getLoggedErrors(): array
+    {
+        $errorLogPath = Config::getInstance()->get('resource.log.error.path');
+
+        $fileHandlerPhp = new FileHandlerPhp();
+
+        if (!$fileHandlerPhp->checkIsPathExists($errorLogPath)) {
+            return [];
+        }
+
+        $errorString = $fileHandlerPhp->getAsString($errorLogPath, 0);
+
+        if (empty($errorString)) {
+            return [];
+        }
+        $errors = [];
+        foreach (explode(PHP_EOL, $errorString) as $error) {
+            $errors[] = json_decode($error, true)['message'];
+        }
+        return $errors;
+    }
+
+    protected function callUri(string $url, string $method = HttpRequest::METHOD_HTTP_GET, array $params = array())
+    {
+        $getParams = [BootstrapAbstract::GET_PARAM_NAME_TESTING_MODE => 1];
+
+        $options = [
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_HEADER         => true,
+            CURLOPT_SSL_VERIFYPEER => false,
+            CURLOPT_SSL_VERIFYHOST => false,
+        ];
+        $this->sentParams = $params;
+
+        switch ($method) {
+            case HttpRequest::METHOD_HTTP_GET:
+                $options[CURLOPT_HTTPGET] = true;
+                $getParams = array_merge($params, $getParams);
+                break;
+
+            case HttpRequest::METHOD_HTTP_POST:
```

Comment:
I agree and understand, I'm just saying your code assumes there are no get params in the URL, because it appends the testing param with a `?`, which will not work if there are any GET params in the URL